include/xtypes/DynamicDataImpl.hpp

```diff
@@ -274,318 +274,424 @@ inline DynamicData DynamicData::operator - () const
-            DYNAMIC_DATA_OPERATOR_RESULT(uint8_t, <<);
+            DYNAMIC_DATA_LOGIC_OPERATION(uint8_t, ||);
         case TypeKind::INT_16_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(int16_t, <<);
+            DYNAMIC_DATA_LOGIC_OPERATION(int16_t, ||);
         case TypeKind::UINT_16_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(uint16_t, <<);
+            DYNAMIC_DATA_LOGIC_OPERATION(uint16_t, ||);
         case TypeKind::INT_32_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(int32_t, <<);
+            DYNAMIC_DATA_LOGIC_OPERATION(int32_t, ||);
         case TypeKind::UINT_32_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(uint32_t, <<);
+            DYNAMIC_DATA_LOGIC_OPERATION(uint32_t, ||);
         case TypeKind::INT_64_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(int64_t, <<);
+            DYNAMIC_DATA_LOGIC_OPERATION(int64_t, ||);
         case TypeKind::UINT_64_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(uint64_t, <<);
+            DYNAMIC_DATA_LOGIC_OPERATION(uint64_t, ||);
+        case TypeKind::FLOAT_32_TYPE:
+            DYNAMIC_DATA_LOGIC_OPERATION(float, ||);
+        case TypeKind::FLOAT_64_TYPE:
+            DYNAMIC_DATA_LOGIC_OPERATION(double, ||);
+        case TypeKind::FLOAT_128_TYPE:
+            DYNAMIC_DATA_LOGIC_OPERATION(long double, ||);
         default:
             xtypes_assert(false,
-                "operator<<() isn't available for types '" << type_.name() << "' and '" << other.type().name() << "'.");
-            return *this;
+                "operator||() isn't available for types '" << type_.name() << " and " <<  other.type().name() << "'.");
     }
+}
+
+#define DYNAMIC_DATA_OPERATOR_RESULT(TYPE, OPERATOR) \
+{\
+    TYPE lho = *this;\
+    TYPE rho = other;\
+    DynamicData result(primitive_type<TYPE>());\
+    result = lho OPERATOR rho;\
+    return result;\
+}
 
+#define DYNAMIC_DATA_NUMERIC_INT_OPERATOR_RESULT(OPERATOR) \
+{\
+    switch(type_.kind())\
+    {\
+        case TypeKind::INT_8_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(int8_t, OPERATOR);\
+        case TypeKind::UINT_8_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(uint8_t, OPERATOR);\
+        case TypeKind::INT_16_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(int16_t, OPERATOR);\
+        case TypeKind::UINT_16_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(uint16_t, OPERATOR);\
+        case TypeKind::INT_32_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(int32_t, OPERATOR);\
+        case TypeKind::UINT_32_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(uint32_t, OPERATOR);\
+        case TypeKind::INT_64_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(int64_t, OPERATOR);\
+        case TypeKind::UINT_64_TYPE:\
+            DYNAMIC_DATA_OPERATOR_RESULT(uint64_t, OPERATOR);\
+        default:\
+            xtypes_assert(false,\
+                "operator" << #OPERATOR << "() isn't available for types '" << type_.name() << "' and '" << other.type().name() << "'.");\
+                return *this;\
+    }\
```

Comment:
Nice! Maybe we can reuse this concept in the other operators to minimize the switches.

Reply:
I've further reused it for other operators in my last commit :slightly_smiling_face: